dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModel.java

```diff
@@ -391,69 +388,75 @@ private void trackDependencies(Event e) {
             curCtrlDeps.removeAll(ifCtrlDeps.pop());
         }
 
-
-        if (e instanceof MemEvent) {
-            // ---- Track address dependency ----
-            MemEvent memEvent = (MemEvent) e;
-            HashSet<EventData> deps = new HashSet<>();
-
-            for (Register reg : memEvent.getAddress().getRegs()) {
-                deps.addAll(lastRegWrites.get(reg));
-            }
-            addrDepMap.put(eventMap.get(e), deps);
-        }
-
         if (e.hasTag(Tag.VISIBLE)) {
             // ---- Track ctrl dependency ----
             // TODO: This may be done more efficiently, as many events share the same set of ctrldeps.
             ctrlDepMap.put(eventMap.get(e), new HashSet<>(curCtrlDeps));
         }
 
-        if (e instanceof ExecutionStatus) {
-            // ---- Track data dependency due to execution tracking ----
-            ExecutionStatus status = (ExecutionStatus) e;
-            Event tracked = status.getStatusEvent();
-            HashSet<EventData> deps = new HashSet<>();
-            if (eventExists(tracked) && status.doesTrackDep()) {
-                deps.add(eventMap.get(tracked));
+        if (e instanceof RegReader regReader) {
+            final Set<EventData> dataDeps = new HashSet<>();
+            final Set<EventData> addrDeps = new HashSet<>();
+            final Set<EventData> ctrlDeps = new HashSet<>();
+            for (Register.Read regRead : regReader.getRegisterReads()) {
+                final Register reg = regRead.register();
+                final Set<EventData> visibleRootDependencies = lastRegWrites.get(reg);
+                if (visibleRootDependencies == null) {
+                    // FIXME: This should never happen, but our parser is buggy and produces ill-formed code.
```

Comment:
This is not the first time we have missing dependencies. We should issue a warning to be sure we don't have other instances (besides the `pthread_join`), and once we get rid of that parser, use an assertion

Reply:
Sure, I will add a `logger.warn`.